cmd/prchecker/main.go

```diff
@@ -19,11 +19,41 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"regexp"
 	"strings"
 
 	"github.com/google/go-github/github"
 )
 
+// splitCommitsHelp is a help message displayed when a PR has commits which
+// span modules.
+const splitCommitsHelp = "\nModifications to multiple restricted directories occurred.\n" +
+	"To resolve this please limit code changes in a commit to a single go module.\n\n" +
+	"You may learn more about how to split your existing commits at: https://git-scm.com/docs/git-rebase#_splitting_commits\n"
+
+// Ignore span pattern defines a regular expression pattern that, when matched,
+// causes this check to be ignored.
+// Pattern expects "ALLOW_MODULE_SPAN" in CAPS on a line by itself.
+// Spaces may be included before or after but no other characters with one
+// exception. ">" may be used to quote the exclusion.
+// Pattern may be provided at any point in the pull request description.
+//
+// Ex: "ALLOW_MODULE_SPAN", "> ALLOW_MODULE_SPAN", "   ALLOW_MODULE_SPAN   "
+const ignoreSpanPattern = "(?m)^>?\\s*ALLOW_MODULE_SPAN\\s*$"
+
+// Changeset represents a set of file modifications associated with a commit
+type Changeset struct {
+	files []string
+	id    *string
```

Comment:
I would avoid having pointers to string types. If the flag library doesn't provide a way to avoid returning those, just check for nil immediately and dereference the pointer. Usually go relies on nil values (which for string is the empty string) instead of a nil pointer to detect the absence of information.